x-pack/plugins/ml/public/file_datavisualizer/components/edit_flyout/overrides.js

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import React, {
+  Component,
+} from 'react';
+
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSuperSelect,
+  EuiCheckbox,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+
+import {
+  formatOptions,
+  timestampFormatOptions,
+  delimiterOptions,
+  quoteOptions,
+  charsetOptions,
+} from './options';
+
+export class Overrides extends Component {
+  constructor(props) {
+    super(props);
+
+    this.originalSettings = this.props.originalSettings;
+    this.overrides = this.props.overrides;
+
+    const {
+      charset,
+      format,
+      hasHeaderRow,
+      columnNames,
+      delimiter,
+      quote,
+      shouldTrimFields,
+      grokPattern,
+      timestampField,
+      timestampFormat,
+    } = this.overrides;
+
+    const {
+      delimiter: d,
+      customDelimiter: customD
+    } = convertDelimiter((delimiter === undefined) ? this.originalSettings.delimiter : delimiter);
+
+    const tempColumnNames = (columnNames === undefined && this.originalSettings.columnNames !== undefined) ?
+      [...this.originalSettings.columnNames] : columnNames;
+
+    this.state = {
+      charset: (charset === undefined) ? this.originalSettings.charset : charset,
+      format: (format === undefined) ? this.originalSettings.format : format,
+      hasHeaderRow: (hasHeaderRow === undefined) ? this.originalSettings.hasHeaderRow : hasHeaderRow,
+      columnNames: tempColumnNames,
+      delimiter: d,
+      customDelimiter: customD,
+      quote: (quote === undefined) ? this.originalSettings.quote : quote,
+      shouldTrimFields: (shouldTrimFields === undefined) ? this.originalSettings.shouldTrimFields : shouldTrimFields,
+      grokPattern: (grokPattern === undefined) ? this.originalSettings.grokPattern : grokPattern,
+      timestampFormat: (timestampFormat === undefined) ? this.originalSettings.timestampFormat : timestampFormat,
+      timestampField: (timestampField === undefined) ? this.originalSettings.timestampField : timestampField,
+    };
+
+    this.fields = this.props.fields;
+    this.fieldOptions = this.fields.map(f => ({ value: f, inputDisplay: f }));
+    this.originalColumnNames = (this.state.columnNames !== undefined) ? [...this.state.columnNames] : [];
```

Comment:
I think the above three lines should go to `render()`. The code might work as intended, but my understanding is that props could change and doing this only in the constructor will do this only once and will not pick up changing props.